internal/translation/teams/teams.go

```diff
@@ -0,0 +1,130 @@
+package teams
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+)
+
+type AtlasTeamsService interface {
+	ListProjectTeams(ctx context.Context, projectID string) ([]Team, error)
+	GetTeamByName(ctx context.Context, orgID, teamName string) (*Team, error)
+	GetTeamByID(ctx context.Context, orgID, teamID string) (*Team, error)
+	Assign(ctx context.Context, at *[]Team, projectID string) error
+	Unassign(ctx context.Context, projectID, teamID string) error
+	Create(ctx context.Context, at *AssignedTeam, orgID string) (*Team, error)
+	RenameTeam(ctx context.Context, at *Team, orgID, newName string) (*Team, error)
+	UpdateRoles(ctx context.Context, at *Team, projectID string, newRoles []akov2.TeamRole) error
+
+	GetTeamUsers(ctx context.Context, orgID, teamID string) ([]TeamUser, error)
+	AddUsers(ctx context.Context, usersToAdd *[]admin.AddUserToTeam, orgID, teamID string) error
+	RemoveUser(ctx context.Context, orgID, teamID, userID string) error
+}
+
+type TeamsAPI struct {
+	teamsAPI     admin.TeamsApi
+	teamUsersAPI admin.MongoDBCloudUsersApi
+}
+
+func NewTeamsAPIService(teamAPI admin.TeamsApi, userAPI admin.MongoDBCloudUsersApi) *TeamsAPI {
+	return &TeamsAPI{
+		teamsAPI:     teamAPI,
+		teamUsersAPI: userAPI,
+	}
+}
+
+func (tm *TeamsAPI) ListProjectTeams(ctx context.Context, projectID string) ([]Team, error) {
+	atlasAssignedTeams, _, err := tm.teamsAPI.ListProjectTeams(ctx, projectID).Execute()
+	if err != nil {
```

Comment:
Why doesn't return AssignedTeam? this relates to the other comment regarding sematics

Reply:
Fixed 👍